### PR TITLE
Support for Error handling 

### PR DESCRIPTION
**Pull request by srmukher**
The following changes have been added:
1. Created a common control that can show error and empty states.
2. Used the common control to show already implemented empty messages.
3. Use the common control to show type errors (eg: when string data is used on x axis for Line chart or area chart which is not supported)
4. Added Error Codes to indicate Generic Error or No Data Error.
5. Kept provision for customizing the error handling while rendering if required.
6. Added support for light/dark theme.

Examples:
1. Generic Types Error 
image

3. Missing Data
image

Monosizes after adding error handling for horizontal bar chart variants:

image

Monosize in master:

image

Bundle size after adding error handling:
image

Bundle size in master:
image

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/ae31ae01c299e43ff98c561bf937da1ed465ef8a'>ae31ae01c299e43ff98c561bf937da1ed465ef8a</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

**Comment by AtishayMsft** [comment]
can you add bundle size increase

File: change/@fluentui-react-charting-e4dc8ea7-f277-4bf1-a526-a85995f9f5e0.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-charting-e4dc8ea7-f277-4bf1-a526-a85995f9f5e0.json:1]: ## [🕵 fluentuiv8 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=30346&commitId=23e53cb0c5612eee8e2631c96d5cc90c8e1d3c2d&env=prod&clientType=FLUENTUI)


  <details><summary><b>react-charting-HorizontalBarChart</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|react-charting-HorizontalBarChart.With_Axis - RTL.default.chromium.png|34479| Changed|
|react-charting-HorizontalBarChart.With_Axis.default.chromium.png|34492| Changed|
|react-charting-HorizontalBarChart.With_Axis - Dark Mode.default.chromium.png|15526| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

File: packages/react-charting/src/components/AreaChart/AreaChart.base.tsx [hunk 2]
```diff
@@ -180,47 +180,49 @@ export class AreaChartBase extends React.Component<IAreaChartProps, IAreaChartSt
...
-                    onMouseOut={this._onRectMouseOut}
-                    onMouseOver={this._onRectMouseMove}
-                  />
-                </g>
-                <g>{this._chart}</g>
-              </>
-            );
-          }}
-        />
+        <div style={{ display: 'flex', justifyContent: 'center', alignItems: 'center', height: '100%' }}>
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/AreaChart/AreaChart.base.tsx:183]: why change in this file

File: change/@fluentui-react-charting-e4dc8ea7-f277-4bf1-a526-a85995f9f5e0.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+  "type": "minor",
+  "comment": "Support for Error handling",
+  "packageName": "@fluentui/react-charting",
```
Comment (AtishayMsft) [inline change/@fluentui-react-charting-e4dc8ea7-f277-4bf1-a526-a85995f9f5e0.json:4]: Test cases and examples are missing

File: packages/react-charting/src/components/CommonComponents/ErrorImage.tsx [hunk 4]
```diff
@@ -0,0 +1,451 @@
+import { ITheme } from '@fluentui/react/lib/Styling';
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/CommonComponents/ErrorImage.tsx:1]: move these files under components/assets folder

File: packages/react-charting/src/components/CommonComponents/ErrorBoundary.tsx [hunk 5]
```diff
@@ -0,0 +1,133 @@
+import * as React from 'react';
+import { getId } from '@fluentui/react/lib/Utilities';
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/CommonComponents/ErrorBoundary.tsx:2]: Add tests for this control.

**Comment by AtishayMsft** [comment]
50kB bump is too much. How can we reduce this.

**Comment by srmukher** [comment]
> can you add bundle size increase

done

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

File: change/@fluentui-react-charting-e4dc8ea7-f277-4bf1-a526-a85995f9f5e0.json (hunk 1 above)
Comment (fabricteam) [inline change/@fluentui-react-charting-e4dc8ea7-f277-4bf1-a526-a85995f9f5e0.json:1]: ## [🕵 fluentuiv8 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=30346&commitId=6b0ba16840e42d48d9265bb81a139c2c1a83219a&env=prod&clientType=FLUENTUI)


  <details><summary><b>react-charting-HorizontalBarChart</b> 11 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|react-charting-HorizontalBarChart.Variant - RTL.default.chromium.png|73895| Changed|
|react-charting-HorizontalBarChart.Variant - Dark Mode.default.chromium.png|31249| Changed|
|react-charting-HorizontalBarChart.Variant.default.chromium.png|75557| Changed|
|react-charting-HorizontalBarChart.With_Benchmark - Dark Mode.default.chromium.png|31968| Changed|
|react-charting-HorizontalBarChart.With_Benchmark - RTL.default.chromium.png|86074| Changed|
|react-charting-HorizontalBarChart.With_Benchmark.default.chromium.png|90375| Changed|
|react-charting-HorizontalBarChart.Basic - RTL.default.chromium.png|151862| Changed|
|react-charting-HorizontalBarChart.Basic - Dark Mode.default.chromium.png|48424| Changed|
|react-charting-HorizontalBarChart.Basic.default.chromium.png|158297| Changed|
|react-charting-HorizontalBarChart.Basic - Dark Mode.hover.chromium.png|0| Removed|
|react-charting-HorizontalBarChart.Basic.hover.chromium.png|0| Removed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

File: change/@fluentui-react-charting-e4dc8ea7-f277-4bf1-a526-a85995f9f5e0.json (hunk 1 above)
Comment (fabricteam) [inline change/@fluentui-react-charting-e4dc8ea7-f277-4bf1-a526-a85995f9f5e0.json:1]: ## [🕵 fluentuiv8 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=30346&commitId=664ead4662f1570b07751a4f5647f3ff800a558e&env=prod&clientType=FLUENTUI)


  <details><summary><b>Pivot - Overflow</b> 4 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Pivot - Overflow.Root.Narrow - Last tab selected.chromium.png|0| Removed|
|Pivot - Overflow.Root.Narrow - Overflow menu.chromium.png|0| Removed|
|Pivot - Overflow.Tabs - RTL.Narrow - Last tab selected.chromium.png|0| Removed|
|Pivot - Overflow.Tabs - RTL.Narrow - Overflow menu.chromium.png|0| Removed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

**Pull request closed without merging**